File: OOPJava/src/main/game/zwierzeta/Wilk.java
package main.game.zwierzeta;
import main.game.*;

public class Wilk extends Zwierze{
    public Wilk(int x, int y, Swiat swiat){ super(9,5,x,y,swiat); }
    public Wilk(int sila, int inicjatywa, int x, int y, int poprzednieX, int poprzednieY, int wiek, Swiat swiat) {
        super(sila, inicjatywa, x, y, poprzednieX, poprzednieY, wiek, swiat);
    };

    @Override
    public String rysowanie(){
        return "wilk.png";
    }
    @Override
    public void rozmnozSie(int x, int y, Swiat swiat){
        Wilk nowyOrganizm = new Wilk(x, y, swiat);
        swiat.dodajOrganizm(nowyOrganizm);
    }

    @Override
    public String getImie() { return "Wilk"; };
}

File: PROJECT_C++/PROJECT_C++/Wilk.cpp
#include "Wilk.h"
#include <iostream>
#include <string>
#include <vector>
#include "Swiat.h"

Wilk::Wilk(int x, int y, Swiat& swiat) : Zwierze(9, 5, x, y, swiat) {};

void Wilk::rysowanie() {
	std::cout << "W";
}

void Wilk::rozmnozSie(int x, int y, Swiat& swiat) {
	Wilk* nowyOrganizm = new Wilk(x, y, swiat);
	swiat.dodajOrganizm(*nowyOrganizm);
}

Wilk::~Wilk() {};